jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskcomments/TaskCommentsPresenter.java

```diff
@@ -87,81 +78,79 @@ public IsWidget getView() {
+                    @Override
+                    public void callback(Long response) {
+                        refreshComments();
+                        view.clearCommentInput();
+                    }
+                },
+                new ShowErrorInModalCallback()
+        ).addComment(currentTaskId, text, identity.getIdentifier(), addedOn);
     }
 
-    public void removeTaskComment( long commentId ) {
-        taskCommentsServices.call( new RemoteCallback<Long>() {
-            @Override
-            public void callback( Long response ) {
-                refreshComments();
-                view.getNewTaskCommentTextArea().setText( "" );
-                view.displayNotification( "Comment Deleted!" );
-            }
-        }, new ErrorCallback<Message>() {
-            @Override
-            public boolean error( Message message,
-                                  Throwable throwable ) {
-                ErrorPopup.showMessage( "Unexpected error encountered : " + throwable.getMessage() );
-                return true;
-            }
-        } ).deleteComment( currentTaskId, commentId );
+    public void removeTaskComment(long commentId) {
+        taskCommentsServices.call(
+                new RemoteCallback<Long>() {
+                    @Override
+                    public void callback(Long response) {
+                        refreshComments();
+                        view.clearCommentInput();
+                        view.displayNotification("Comment Deleted!");
+                    }
+                },
+                new ShowErrorInModalCallback()
+        ).deleteComment(currentTaskId, commentId);
     }
 
-    public void addDataDisplay( final HasData<CommentSummary> display ) {
-        dataProvider.addDataDisplay( display );
+    public void addDataDisplay(final HasData<CommentSummary> display) {
+        dataProvider.addDataDisplay(display);
     }
 
-    public void onTaskSelectionEvent( @Observes final TaskSelectionEvent event ) {
-        this.currentTaskId = event.getTaskId();
+    public void onTaskSelectionEvent(@Observes final TaskSelectionEvent event) {
+        currentTaskId = event.getTaskId();
         refreshComments();
-        view.getDataGrid().redraw();
+        view.redrawDataGrid();
     }
 
-    public void onTaskRefreshedEvent( @Observes final TaskRefreshedEvent event ) {
-        if ( currentTaskId == event.getTaskId() ) {
+    public void onTaskRefreshedEvent(@Observes final TaskRefreshedEvent event) {
+        if (currentTaskId == event.getTaskId()) {
             refreshComments();
-            view.getDataGrid().redraw();
+            view.redrawDataGrid();
+        }
+    }
+
+    private static class ShowErrorInModalCallback implements ErrorCallback<Message> {
+
+        @Override
+        public boolean error(Message message, Throwable throwable) {
+            ErrorPopup.showMessage("Unexpected error encountered : " + throwable.getMessage());
+            return true;
```

Comment:
There's this (but it'll only show a "generic" error message):

https://github.com/uberfire/uberfire-extensions/blob/master/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/callbacks/DefaultErrorCallback.java

Or there's this (that you can customize to perform different actions for
different Throwables):

https://github.com/droolsjbpm/kie-wb-common/blob/master/kie-wb-common-widgets/kie-wb-common-ui/src/main/java/org/kie/workbench/common/widgets/client/callbacks/CommandDrivenErrorCallback.java

The latter extends ```HasBusyIndicatorDefaultErrorCallback``` (so your View
would need to implement HasBusyIndicator).


On 9 July 2015 at 10:19, Salaboy <notifications@github.com> wrote:

> In
> jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskcomments/TaskCommentsPresenter.java
> <https://github.com/droolsjbpm/jbpm-console-ng/pull/167#discussion_r34237094>
> :
>
> >              refreshComments();
> > -            view.getDataGrid().redraw();
> > +            view.redrawDataGrid();
> > +        }
> > +    }
> > +
> > +    private static class ShowErrorInModalCallback implements ErrorCallback<Message> {
> > +
> > +        @Override
> > +        public boolean error(Message message, Throwable throwable) {
> > +            ErrorPopup.showMessage("Unexpected error encountered : " + throwable.getMessage());
> > +            return true;
>
> +1 to @jhrcek <https://github.com/jhrcek> comment below, we need a
> generic class to use in all the error callbacks. Probably there is already
> one in Uberfire Extension that we can use.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/droolsjbpm/jbpm-console-ng/pull/167/files#r34237094>.
>